HW04.Rmd

```diff
@@ -77,45 +175,217 @@ Assume that:
+levels(deaths_police$race)[levels(deaths_police$race) == "Hispanic"] <-"His"
+levels(deaths_police$race)[levels(deaths_police$race) == "Not Availabletive American/Not Availabletive Alaskan"] <-"NNA"
+levels(deaths_police$race)[levels(deaths_police$race) == "White"] <-"WH"
+
+#Police involvement vs. race.
+dp_race <- deaths_police %>%
+    select(police, race) %>%
+    group_by(police) %>%
+    count(race) %>%
+    arrange(race)
+
+#Police involvement vs. age.
+dp_age <- deaths_police %>%
+    select(police, age)
+
+kable(dp_race, format = "markdown", col.names = c("Police involvement", "Race", "Number of deaths"))
+
+dp_race %>%
+    ggplot(aes(x = race, y = n, fill = police)) +
+    geom_col(position = "dodge") +
+    scale_y_log10() +
+    annotation_logticks(sides = "l") +
+    labs(title = "Police involvement in gun deaths according to race",
+         x = "Race",
+         y = "Number of deaths (log 10)",
+         fill = "Police involvement",
+         caption = "Source: fivethirtyeight.com")  
+
+dp_race_sum <- dp_race %>%
+    summarize(sum_v = sum(n), round(per_v = (sum_v * 100) / 35167), mean_v = mean(n))
+
+kable(dp_race_sum, format = "markdown", col.names = c("Police involvement", "Total gun deaths", "Percentage (%)", "Average deaths among all races"))
 
 ```
 
+**Answer**
+As seen before, suicide represents an important portion of the database and can bias the analysis of police involvement, so the data was filtered to only include homicide.
+
+About police involvement and race, it seems that police officers are involved in only 4% of homicides in the United States (see table). When evaluated by race, involvement is generally low too. If we consider the average deaths among all races, black and white populations are the most affected by police involvement, while hispanic population are right on the average value (see table). 
+
 Think about performing stats such as a t.test to see if differences you are plotting are statistically significant
 
 ```{r stats?}
 
+#It's been a while since I had my statistics course, so if this analysis is wrong in any way, feel free to explain to me how to correct it. 
+
+
+#Kolmogorov-Smirnov test to check for normality in the data
+ks.test(dp_age$age[dp_age$police == "Not involved"],
+       dp_age$age[dp_age$police == "Involved"])
+
+#Fligner-Killeen test to check for heteroscedasticity
+fligner.test(list(dp_age$age[dp_age$police == "Not involved"],
+       dp_age$age[dp_age$police == "Involved"]))
+
+#Mann-Whitman-Wilcoxon test to compare non-normal samples
+wilcox.test(dp_age$age[dp_age$police == "Not involved"],
+                       dp_age$age[dp_age$police == "Involved"])
+
 
 ```
 
 
-```{r police_2}
+```{r police_2, echo=F}
+
+dp_age %>%
+    ggplot(aes(x = police, y = age, fill = police)) +
+    geom_boxplot(alpha = 0.5) +
+    geom_signif(comparisons = list(c("Not involved", "Involved")), map_signif_level = TRUE, test = wilcox.test) +

```

Comment:
Love to see this come in! I've always seen this notation on graphs and now I know how to do it, thanks for sharing!

Reply:
Me too! I always saw that notation on papers; it was amazing to discover how to do it in R.